src/main/java/io/quarkus/security/identity/request/BaseAuthenticationRequest.java

```diff
@@ -0,0 +1,27 @@
+package io.quarkus.security.identity.request;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Base implementation of the {@link AuthenticationRequest} interface for convenience.
+ */
+public abstract class BaseAuthenticationRequest implements AuthenticationRequest {
+
+    private final Map<String, Object> attributes = new HashMap<>();
```

Comment:
I would allocate this in a lazy manner, to avoid additional allocations if the map is not used.

Reply:
@stuartwdouglas  Done. Let me know if you want further adjustments.

Reply:
@stuartwdouglas Is it okay to set this PR to ready for review?